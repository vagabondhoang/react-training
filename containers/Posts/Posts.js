// @flow

import React from 'react';
import { connect } from 'react-redux';

import { ListPost } from '../../components/ListPost';

import getListing from './selectors';

type Props = {
  posts: Array<Object>,
};

const Posts = ({ posts }: Props) => (
  <div>
    <h3>Posts</h3>
    <ul>
      <ListPost posts={posts} />
    </ul>
  </div>
);

const mapStateToProps = state => ({
  posts: getListing(state),
});

export default connect(mapStateToProps)(Posts);
