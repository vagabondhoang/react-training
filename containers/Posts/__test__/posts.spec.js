import React from 'react';
import Enzyme, { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';

import Posts from '../Posts';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();

describe('post container', () => {
  const mockedState = {
    postsById: {
      postsById: {
        postsById: {},
      },
    },
    usersById: {
      usersById: {
        usersById: {},
      },
    },
    postListing: {
      postListing: [],
    },
  };

  const store = mockStore(mockedState);

  const component = mount(<Posts store={store} />);

  it('render correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
