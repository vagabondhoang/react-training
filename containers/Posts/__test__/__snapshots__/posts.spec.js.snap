// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post container render correctly 1`] = `
<Connect(Posts)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Posts
    dispatch={[Function]}
    posts={Array []}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div>
      <h3>
        Posts
      </h3>
      <ul>
        <ListPost
          posts={Array []}
        />
      </ul>
    </div>
  </Posts>
</Connect(Posts)>
`;
